spec/controllers/users_controller_spec.rb

```diff
@@ -76,44 +75,75 @@
 
   describe 'POST #create' do
     let(:account) { create :account }
+    let(:admin_user) { create :user }
     let(:user) { attributes_for :user }
-    before do
-      user.merge!(account_id: account.id)
-      post :create, params: { user: user }
-    end
-
-    it 'returns http success' do
-      expect(response).to have_http_status(:success)
-    end
-
-    it 'JSON body response match account attributes' do
-      user_responded = response.parsed_body
-      expect(JSON::Validator.fully_validate(user_schema, user_responded)).to be_empty
+    context 'when user is authenticated' do
+      include_context 'with authenticated user'
+      before do
+        user.merge!(account_id: account.id)
+        request.headers.merge! default_headers
+        post :create, params: { user: user }
+      end
+
+      it 'returns http success' do
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'JSON body response match account attributes' do
+        expect(response).to match_json_schema(user_schema)
+      end
+    end
+    context 'when user is not authenticated' do
+      before do
+        user.merge!(account_id: account.id)
```

Comment:
If you put the before block outside these contexts, you can just overwrite the `let(:default_headers) { ...fake_or_empty_auth_headers }` here and let the before block be the same for the whole spec